spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkScan.java

```diff
@@ -911,9 +1027,17 @@ private void checkColStatisticsReported(
     assertThat(stats.numRows().getAsLong()).isEqualTo(expectedRowCount);
 
     Map<NamedReference, ColumnStatistics> columnStats = stats.columnStats();
-    for (Map.Entry<String, Long> entry : expectedNDVs.entrySet()) {
-      assertThat(columnStats.get(FieldReference.column(entry.getKey())).distinctCount().getAsLong())
-          .isEqualTo(entry.getValue());
+    if (expectedNDVs.isEmpty()) {
+      assertThat(
+              columnStats.isEmpty()
+                  || columnStats.values().iterator().next().distinctCount().isEmpty())
```

Comment:
I'm not sure I understand the second check here. Shouldn't we be checking all `columnStats.values().distinctCount` are empty?

Reply:
Hi @RussellSpitzer the check will not work like this, if we do columnStats.values().distinctCount, it will give error as "Cannot resolve method 'distinctCount' in 'Collection'"

Reply:
Understood, I have updated it, could you please review again.